packages/rust/server/tests/message_test.rs

```diff
@@ -0,0 +1,328 @@
+    };
+    let res = client
+        .post("/message")
+        .header("Content-Type", "application/json")
+        .body(serde_json::to_string(&message_2).unwrap())
+        .header("Authorization", bearer.clone())
+        .send()
+        .await;
+
+    assert_eq!(res.status(), StatusCode::OK);
+
+    let message_3 = MessageRequest {
+        client_uuids: vec![client_uuid.to_string()],
+        message: Base64("test message 3".as_bytes().to_vec()),
+    };
+    let res = client
+        .post("/message")
+        .header("Content-Type", "application/json")
+        .body(serde_json::to_string(&message_3).unwrap())
+        .header("Authorization", bearer.clone())
+        .send()
+        .await;
+
+    assert_eq!(res.status(), StatusCode::OK);
+
+    let request_messages = CheckMessages {
+        client_uuid: client_uuid.to_string(),
+    };
+    let res = client
+        .get("/message")
+        .header("Content-Type", "application/json")
+        .body(serde_json::to_string(&request_messages).unwrap())
+        .header("Authorization", bearer.clone())
+        .send()
+        .await;
+
+    assert_eq!(res.status(), StatusCode::OK);
+    let messages = res.json::<MessagesReturned>().await.messages;
+    assert_eq!(messages.len(), 3);
+    assert!(messages.contains(&message_1.message));
+    assert!(messages.contains(&message_2.message));
+    assert!(messages.contains(&message_3.message));
+}
+
+#[tokio::test]
+async fn test_bad_uuid() {
+    let db = TempDatabase::new().await;
+    let client = start_server(db.pool().clone()).await;
+
+    let created_user = CreateUser {
+        email: "test@gmail.com".to_string(),
+        username: "test_username".to_string(),
+        password: "test_password".to_string(),
+        name: "test_name".to_string(),
+        identity: Base64(PUBLIC.to_vec()),
+    };
+    let (token, _) = helper::initialize_user(db.pool(), &client, &created_user)
+        .await
+        .unwrap();
+
+    let bearer = format!("Bearer {}", token);
+    let (_, client_uuid) = helper::create_client(PUBLIC, PRIVATE, &bearer, &client).await;
+
+    let bad_uuid = Uuid::new_v4().to_string();
+
+    let request_messages = CheckMessages {
+        client_uuid: bad_uuid.clone(),
+    };
+    let res = client
+        .get("/message")
+        .header("Content-Type", "application/json")
+        .body(serde_json::to_string(&request_messages).unwrap())
+        .header("Authorization", bearer.clone())
+        .send()
+        .await;
+
+    assert_eq!(res.status(), StatusCode::NOT_FOUND);
+
+    let request_messages = CheckMessages {
+        client_uuid: "bad uuid".to_string(),
+    };
+    let res = client
+        .get("/message")
+        .header("Content-Type", "application/json")
+        .body(serde_json::to_string(&request_messages).unwrap())
+        .header("Authorization", bearer.clone())
+        .send()
+        .await;
+
+    assert_eq!(res.status(), StatusCode::BAD_REQUEST);
```

Comment:
Given that serde handles parsing for us this is actually just testing serde, not our code